modules/phase_unwrapping/include/opencv2/phase_unwrapping/histogramphaseunwrapping.hpp

```diff
@@ -92,6 +92,11 @@ class CV_EXPORTS_W HistogramPhaseUnwrapping : public PhaseUnwrapping
     static Ptr<HistogramPhaseUnwrapping> create( const HistogramPhaseUnwrapping::Params &parameters =
                                                  HistogramPhaseUnwrapping::Params() );
 
+    // alias for scripting
+    CV_WRAP
+    static Ptr<HistogramPhaseUnwrapping> create( int width, int height, float histThresh = static_cast<float>(3 * CV_PI * CV_PI),
+                                                 int nbrOfSmallBins = 10, int nbrOfLargeBins = 5);
```

Comment:
Please follow the existed practice and avoid unpacking of "Params" structure.
You can check this code as an example:

https://github.com/opencv/opencv/blob/850414a501d5e6dba111c92d73fdd05794c7061d/modules/features2d/include/opencv2/features2d.hpp#L653-L654

Reply:
The existing practice is not clear as the structured_light itself is using 2 practices:
- Params unpacking in `GrayCodePattern::create`
- smart pointer for `SinusoidalPattern::create`

Some codes pass a smart pointer like this is done for `SinusoidalPattern::create`

https://github.com/opencv/opencv_contrib/blob/master/modules/structured_light/include/opencv2/structured_light/sinusoidalpattern.hpp#L100

 but I followed the solution used by `GrayCodePattern::create`

https://github.com/opencv/opencv_contrib/blob/master/modules/structured_light/include/opencv2/structured_light/graycodepattern.hpp#L89

Because changing the create API (instead of overloading it) will break current codes using structured_light.

The code you point out is not using Params struct to pass parameters.

If you just pass a `const Params&`, the Python binding generator will not generate the need code. Am I right ?

Reply:
"Params" class should be marked for bindings wrapping too (use `CV_EXPORTS_W_SIMPLE`)

Reply:
OK. I managed to bind the `HistogramPhaseUnwrapping::Params`; What I was missing is the `typedef` in `pyopencv_phase_unwrapping.hpp` as it is done in `pyopencv_features2d.hpp`

I added the Python test for structured_light replicating what is done in C++. However, this shown a "bug" in `HistogramPhaseUnwrapping::unwrapPhaseMap` where the `unwrappedPhaseMap` were uninitialized in places outside of the mask.
It is not clear if this were intentional or not.